.github/workflows/ci.yml

```diff
@@ -6,8 +6,30 @@ env:
   GRADLE_OPTS: -Dhttp.keepAlive=false
 
 jobs:
+  generate-test-list:
+    runs-on: ubuntu-latest
+    outputs:
+      separateTestsNames: ${{ steps.set-matrix.outputs.separateTestsNames }}
+      testsNamesExclude: ${{ steps.set-matrix.outputs.testsNamesExclude }}
+    steps:
+    - name: Set up JDK for build and test
+      uses: actions/setup-java@v2
+      with:
+        distribution: temurin # Temurin is a distribution of adoptium
+        java-version: 17
+
+    - name: Checkout security
+      uses: actions/checkout@v2
+
+    - name: Generate list of tasks
+      id: set-matrix
+      run: |
+        echo "separateTestsNames=$(./gradlew listTasksAsJSON -q --console=plain | tail -n 1)" >> $GITHUB_OUTPUT
+        echo "testsNamesExclude=$(./gradlew listTasksAsParam -q --console=plain | tail -n 1)" >> $GITHUB_OUTPUT
```

Comment:
I'd really prefer we didn't have these exclusions as it spreads the responsiblity of executing the tests based on a matrix of tests and several gradle task, can we see about removing them and grouping all the tests together into a single test task?

Reply:
Apologies, but I don't really understand this. Can you elaborate more?

If we had all tests in a single task, then how shall we achieve the parallel execution in `workflows/ci.yml`?

Reply:
@peternied I'm not sure whether you understand what is the point of this PR or even what was the problem in the first place. Please let me know if you need more information.

> Trigger a single test. This is broken today because the openssl tasks run even with the filter condition. I understand if this is out of scope and I'll write up a separate issue

This is broken and will become even more broken after this PR. It looks like this is broken by design because (old) _opensslTest_ was dependency of "test" task so it was always started when "test" was run. This PR will make this issue BIGGER because all separated tests are dependencies.

Simplest way to fix is the same as within old one - just _-x_ tasks which you do not want to run. This PR gives you a simple method:

`./gradlew test --tests org.opensearch.security.my.test $(./gradlew listTasksAsParam -q --console=plain | tail -n 1)`

> Trigger a subset of the tests as if I was a github action runner: ./gradlew dlicDlsflsTest - Pretty sure this is what you are doing now, but when I look at the way the gradle task is constructed there is alot of -x'ed items

We must exclude tests which are started from "test" task because we do not want them to be run twice. The point of this PR is to shorten time for test run on Github Actions and that would be pointless if we:
1. Start test from `org.opensearch.security.dlic.*`  by running `gradlew build dlicDlsflsTest`
2. Run them AGAIN as a part of `gradlew build test` command.

Reply:
@peternied 

That sounds very reasonable to me, we are looking into this at the moment.

I just have one question: We are a bit confused about the openssl test - why was it split into a separate task? It seems to complicate the management of the tests a bit. Do you have any preferences on how to handle this test?

Reply:
With 33f8318 adding the OpenSSL tests into their own execution I'm going to resolve this thread as there is nothing I see for follow up